### DISCUSSION TITLE
How to use lighting effects on 3d tiles?

### DISCUSSION THREAD

**Discussion by chrisgervang** [visgl/deck.gl#8095] [Q&A]
I've noticed lighting effects don't work on the `ScenegraphLayer`s inside `Tile3DLayer`. I'm using google 3d tiles, but I don't think that's relevant. A scenegraph layer outside of tiles works fine with lighting. Any ideas why that might be?

```js
const layers = [
  new Tile3DLayer({
    id: '3d-tiles',
    data: ...,
    _subLayerProps: {
      scenegraph: {_lighting: 'pbr'} // pbr is the default
    }
  }),
  new ScenegraphLayer({
    id: 'scenegraph',
    ...
    _lighting: 'pbr'
  })
]

const ambientLight = new AmbientLight({
  color: [255, 255, 255]
  intensity: 0,
})

const directionalLight = new DirectionalLight({
  color: [255, 0, 0], // expecting red hue on all layers
  intensity: 16,
  direction: [1, -1, 0.5],
})

const lightingEffect = new LightingEffect({
  ambientLight,
  directionalLight,
})
```

<img width="948" alt="Screen Shot 2023-09-03 at 8 02 43 AM" src="https://github.com/visgl/deck.gl/assets/2461547/4302e057-5cb2-4a98-bd3d-4dfa5a093ef6">

**Comment by ibgreen** [comment]
My current understanding is that the geometries provided by Google 3D Tiles do not include normals, which will effectively defeat lighting. 

@igorDykhta was suggesting that we use shader derivatives to approximate normals. 

We do use shader derivates in some other cases, to good effect.

So it should be relatively straightforward, but we haven't been able to get to it yet.

It could be valuable if someone was able to confirm with Google what the situation / their plans around normals are.
